radix-engine/src/kernel/call_frame.rs

```diff
@@ -484,6 +536,150 @@ impl<C, L: Clone> CallFrame<C, L> {
         &self.call_frame_data
     }
 
+    pub fn create_node<'f, S: SubstateStore, E>(
+        &mut self,
+        substate_io: &mut SubstateIO<S>,
+        handler: &mut impl CallFrameEventHandler<C, L, E>,
+        node_id: NodeId,
+        node_substates: NodeSubstates,
+    ) -> Result<(), CallbackError<CreateNodeError, E>> {
+        // TODO: We need to protect transient blueprints from being globalized directly
+        // into store. This isn't a problem for now since only native objects are allowed
+        // to be transient.
+
+        let destination_device = if node_id.is_global() {
+            SubstateDevice::Store
+        } else {
+            SubstateDevice::Heap
+        };
+
+        let mut move_nodes_from_heap = index_set_new();
+        let mut add_non_global_refs = index_map_new();
+
+        // FIXME: Should process all owns before all references to ensure owns don't clash with references
```

Comment:
This is actually a newly found existing bug :) I don't think it actually allows creation of duplicated owns but rather allows you store references to nodes which are owned in the same substate.

Reply:
I think you're right about owns, because we take nodes in `process_substate`. 

I guess I was more concerned about the combining references from multiple substates into a single set, which may break ref counting (which is done at substate level)?